src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/SemanticModel/PXGraph/Enums/GraphSemanticModelCreationOptions.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace Acuminator.Utilities.Roslyn.Semantic.PXGraph
+{
+	/// <summary>
+	/// Values that represent graph semantic model creation options.
+	/// </summary>
+	[Flags]
+	public enum GraphSemanticModelCreationOptions
+	{
+		/// <summary>
+		/// A binary constant representing the collect general graph Information flag.
+		/// </summary>
+		CollectGeneralGraphInfo = 0,
+	
+		/// <summary>
+		/// A binary constant representing the collect processing delegates flag.
+		/// </summary>
+		CollectProcessingDelegates = 0b_00001,
+		
+		/// <summary>
+		/// A binary constant representing the infer implicit models flag.
+		/// </summary>
+		InferImplicitModels = 0b_00010,
+
+		/// <summary>
+		/// Collect all possible semantic data during construction of the graph semantic model.
+		/// </summary>
+		CollectAll = CollectGeneralGraphInfo | CollectProcessingDelegates | InferImplicitModels
+	}
+
+
+	public static class GraphSemanticModelCreationOptionsUtils
+	{
+		public static bool HasOption(this GraphSemanticModelCreationOptions creationOptions, GraphSemanticModelCreationOptions flagToCheck) =>
```

Comment:
This method can be replaced with `Enum.HasFlag`

Reply:
Yes but it will cause boxing
https://stackoverflow.com/questions/11665279/why-enums-hasflag-method-need-boxing

Reply:
Well in general I disagree - we should try to decrease a number of heap allocations .VS is slow enough by itself. However, in this case it indeed doesn't look like a hot path since code map is shared between analyzers and is created only once both for analysis, navigation and code map

Reply:
fixed